opm/simulators/linalg/ISTLSolverEbos.hpp

```diff
@@ -645,29 +646,112 @@ class WellModelMatrixAdapter : public Dune::AssembledLinearOperator<M,X,Y>
         }
 
         /// Zero out off-diagonal blocks on rows corresponding to overlap cells
-        /// Diagonal blocks on ovelap rows are set to diag(1e100).
+        /// Diagonal blocks on ovelap rows are set to diag(1.0).
         void makeOverlapRowsInvalid(Matrix& ebosJacIgnoreOverlap) const
         {
             //value to set on diagonal
             Dune::FieldMatrix<Scalar, numEq, numEq> diag_block(0.0);
             for (int eq = 0; eq < numEq; ++eq)
-                diag_block[eq][eq] = 1.0e100;
+                diag_block[eq][eq] = 1.0;
 
             //loop over precalculated overlap rows and columns
-            for (auto row = overlapRowAndColumns_.begin(); row != overlapRowAndColumns_.end(); row++ )
+            for (auto row = overlapRows_.begin(); row != overlapRows_.end(); row++ )
             {
-                int lcell = row->first;
-                //diagonal block set to large value diagonal
+                int lcell = *row;
+                // Zero out row.
+                ebosJacIgnoreOverlap[lcell] = 0.0;
+
+                //diagonal block set to diag(1.0).
                 ebosJacIgnoreOverlap[lcell][lcell] = diag_block;
+            }
+        }
+
+        /// Create sparsity pattern of matrix without off-diagonal ghost entries.
+        void noGhostAdjacency()
+        {
+            auto grid = simulator_.vanguard().grid();
+            typedef typename Matrix::size_type size_type;
+            size_type numCells = grid.numCells();
+            noGhostMat_.reset(new Matrix(numCells, numCells, Matrix::random));
 
-                //loop over off diagonal blocks in overlap row
-                for (auto col = row->second.begin(); col != row->second.end(); ++col)
+            std::vector<std::set<size_type>> pattern;
+            pattern.resize(grid.numCells());
+
+            auto lid = grid.localIdSet();   
```

Comment:
I think we should resort to using an appropriate mapper instead:
```
typedef typename GET_PROP_TYPE(TypeTag, ElementMapper) ElementMapper;
#if DUNE_VERSION_NEWER(DUNE_GRID, 2,6)
ElementMapper elemMapper(gridView, Dune::mcmgElementLayout());
#else
ElementMapper elemMapper(gridView);
#endif
...
auto idx = elemMapper.index(elem);
```
For Cpgrid this is hopefully the same but for others it might differ and the original lid might not be consecutive.